openstack/loadbalancer/v2/flavorprofiles/requests.go

```diff
@@ -0,0 +1,110 @@
+package flavorprofiles
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// LIST
+
+type ListOptsBuilder interface {
+	ToFlavorProfileListQuery() (string, error)
+}
+
+type ListOpts struct {
+	Fields []string `q:"fields"`
+}
+
+func (opts ListOpts) ToFlavorProfileListQuery() (string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	return q.String(), err
+}
+
+func List(c *gophercloud.ServiceClient, opts ListOptsBuilder) pagination.Pager {
+	url := rootURL(c)
+	if opts != nil {
+		query, err := opts.ToFlavorProfileListQuery()
+		if err != nil {
+			return pagination.Pager{Err: err}
+		}
+		url += query
+	}
+	return pagination.NewPager(c, url, func(r pagination.PageResult) pagination.Page {
+		return FlavorProfilePage{pagination.LinkedPageBase{PageResult: r}}
+	})
+}
+
+// CREATE
+
+type CreateOptsBuilder interface {
+	ToFlavorProfileCreateMap() (map[string]interface{}, error)
+}
+
+type CreateOpts struct {
+	Name         string `json:"name" required:"true"`
+	ProviderName string `json:"provider_name" required:"true"`
+	FlavorData   string `json:"flavor_data" required:"true"`
+}
+
+func (opts CreateOpts) ToFlavorProfileCreateMap() (map[string]interface{}, error) {
+	return gophercloud.BuildRequestBody(opts, "flavorprofile")
+}
+
+func Create(c *gophercloud.ServiceClient, opts CreateOptsBuilder) (r CreateResult) {
+	b, err := opts.ToFlavorProfileCreateMap()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	resp, err := c.Post(rootURL(c), b, &r.Body, nil)
+	_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)
+	return
+}
+
+// GET
+
+func Get(c *gophercloud.ServiceClient, id string) (r GetResult) {
```

Comment:
According to [the docs](https://docs.openstack.org/api-ref/load-balancer/v2/index.html?expanded=list-flavors-detail%2Cshow-flavor-details-detail#show-flavor-profile-details), `GET` also optionally takes a `fields` parameter, similar to List() above. This is just FYI, we don't have to implement it in this PR.